app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :omniauthable, omniauth_providers: [:google_oauth2]
+
+  class << self
+    def from_omniauth(auth)
+      User.find_or_create_by(provider: auth.provider, uid: auth.uid, email: auth.info.email) do |user|
+        set_user_attributes(user, auth)
+      end
+    end
+
+    private
+
+    def set_user_attributes(user, auth)
+      user.provider = auth.provider
+      user.uid = auth.uid
+
+      profile = auth.info
+      user.email = profile.email
+      user.password = Devise.friendly_token[0, 20]
+      user.full_name = profile.name
+      user.avatar_url = profile.image
```

Comment:
What is this is null? Not all user accounts have a profile image. You might need to use the [safe navigation operator](https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/) if that can cause an error 💡